### ISSUE TITLE
Incorrect handling of async generator functions in async loops 

### ISSUE THREAD

**Issue by cthwaite** [microsoft/pyright#200]
**Describe the bug**
pyright does not appear to recognise the use of asynchronous generator functions in asynchronous loops, as described in [PEP525](https://www.python.org/dev/peps/pep-0525/#asynchronous-generators):

```python
async def g1():
    yield 1
    yield 2

async def g2():
    async for v in g1():
        yield v
```

When an async generator function is used in an async loop, pyright raises the following error:
```
'Awaitable[T]' is not iterable
'__aiter__' method not defined
```
Where the type `T` will depend on whether or not the async generator function is annotated (see below)

**To Reproduce**
This reproduction uses [the example given in PEP525](https://www.python.org/dev/peps/pep-0525/#asynchronous-yield-from).
1. Define an async generator function, consisting of "an async function containing one or more `yield`expressions". Optionally [annotate the generator function](https://docs.python.org/3/library/typing.html#typing.AsyncGenerator) with a return value of `AsyncGenerator`.
```python
async def g1() -> AsyncGenerator[int, None]:
    yield 1
    yield 2
```
2. Use the async generator function in an async loop:
```python
async def g2():
    async for v in g2():
        yield v
```
3. Run pyright. Without return type annotation:
```
error: 'Awaitable[Generator[Union[Literal[2], Literal[1]], Any, Any]]' is not iterable
'__aiter__' method not defined
```
With return type annotation:
```
error: 'Awaitable[AsyncGenerator[int, None]]' is not iterable
'__aiter__' method not defined
```

**Expected behavior**
pyright should not raise an error

**Screenshots or Code**
The following working example is evaluated without error by both mypy and pytype, where pyright raises the error described above: 
```python
import asyncio
from typing import AsyncGenerator

async def gen() -> AsyncGenerator[int, None]:
    yield 1
    yield 2

async def main():
    async for v in gen():
        print(v)

asyncio.get_event_loop().run_until_complete(main())
```

**VS Code extension or command-line**
- Command line, version 1.0.55
- Mac OS X 10.12.6
- No config file

**Comment by erictraut** [comment]
Thanks for the bug report. This is indeed a bug. A fix will be included in the next published version of pyright.

**Comment by cthwaite** [comment]
Great - many thanks for the swift reply!

**Comment by erictraut** [comment]
This is now fixed in 1.0.56, which I just published.
